apps/drec-api/src/pods/organization/organization.service.ts

```diff
@@ -66,4 +58,113 @@ export class OrganizationService {
+
+    this.logger.debug(
+      `Successfully registered a new organization with id ${organizationToRegister.code}`,
+    );
+
+    return stored;
+  }
+
+  public async create(
+    organizationToRegister: NewOrganizationDTO,
+  ): Promise<Organization> {
+    this.logger.debug(
+      `Requested organization registration ${JSON.stringify(
+        organizationToRegister,
+      )}`,
+    );
+
+    const allOrganizationsCount = await this.repository.count();
+    const blockchainAccountAddress = await this.generateBlockchainAddress(
+      allOrganizationsCount,
+    );
+    const organizationToCreate = new Organization({
+      ...organizationToRegister,
+      blockchainAccountAddress,
+    });
+
+    const stored = await this.repository.save(organizationToCreate);
+
+    this.logger.debug(
+      `Successfully registered a new organization with id ${organizationToRegister.code}`,
+    );
+
+    return stored;
+  }
+
+  private async generateBlockchainAddress(index: number): Promise<string> {
+    const issuerAccount = Wallet.fromMnemonic(
+      process.env.MNEMONIC!,
+      `m/44'/60'/0'/0/${0}`,
+    ); // Index 0 account
+
+    const [primaryRpc, fallbackRpc] = process.env.WEB3!.split(';');
+    const provider = getProviderWithFallback(primaryRpc, fallbackRpc);
+    const blockchainAccount = Wallet.fromMnemonic(
+      process.env.MNEMONIC!,
+      `m/44'/60'/0'/0/${index + 1}`,
+    );
+
+    const blockchainProperties = await this.blockchainPropertiesService.get();
+
+    const registryWithSigner = Contracts.factories.RegistryFactory.connect(
+      blockchainProperties?.registry,
+      new Wallet(blockchainAccount.privateKey, provider),
+    );
+
+    await registryWithSigner.setApprovalForAll(issuerAccount.address, true);
+
+    return blockchainAccount.address;
+  }
+
+  async update(
+    updateOrganizationDTO: UpdateOrganizationDTO,
+  ): Promise<Organization> {
+    const currentOrg = await this.findOne(updateOrganizationDTO?.code);
+    if (!currentOrg) {
+      throw new NotFoundException(
+        `No organization found with code ${updateOrganizationDTO?.code}`,
+      );
+    }
+    const {
+      name,
+      address,
+      primaryContact,
+      telephone,
+      email,
+      regNumber,
+      vatNumber,
+      regAddress,
+      country,
+      role,
+    } = updateOrganizationDTO;
+    currentOrg.name = name || currentOrg.name;
+    currentOrg.address = address || currentOrg.address;
+    currentOrg.primaryContact = primaryContact || currentOrg.primaryContact;
+    currentOrg.telephone = telephone || currentOrg.telephone;
+    currentOrg.email = email || currentOrg.email;
+    currentOrg.regNumber = regNumber || currentOrg.regNumber;
+    currentOrg.vatNumber = vatNumber || currentOrg.vatNumber;
+    currentOrg.regAddress = regAddress || currentOrg.regAddress;
+    currentOrg.country = country || currentOrg.country;
+    currentOrg.role = role || currentOrg.role;
```

Comment:
you can use https://lodash.com/docs/4.17.15#defaults

Reply:
`_.defaults(updateOrganizationDTO, currentOrg)`

Reply:
Cool. The only issue I have with this is that there are certain properties which should not be changed. So if someone would send the complete organization instead of our Dto, then they can try to change the code of the organization.

Reply:
So you should create proper interfaces and validation on api.

Code will not change anyway, because you find organization by code.